plugins/devtools-backend/src/api/DevToolsBackendApi.ts

```diff
@@ -0,0 +1,222 @@
+      'devTools.externalDependencies.endpoints',
+    );
+    if (!endpoints) {
+      // No external dependency endpoints configured
+      return result;
+    }
+    for (const endpoint of endpoints) {
+      this.logger?.info(
+        `Checking external dependency "${endpoint.name}" at "${endpoint.target}"`,
+      );
+
+      switch (endpoint.type) {
+        case 'ping': {
+          const pingResult = await this.pingExternalDependency(endpoint);
+          result.push(pingResult);
+          break;
+        }
+        case 'fetch': {
+          const fetchResult = await this.fetchExternalDependency(endpoint);
+          result.push(fetchResult);
+          break;
+        }
+        default:
+          return result;
+      }
+    }
+
+    return result;
+  }
+
+  private async fetchExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    let status;
+    let error;
+
+    await fetch(endpoint.target)
+      .then(res => {
+        status = res.status === 200 ? 'Healthy' : 'Unhealthy';
+        this.logger.info(
+          `Fetch for ${endpoint.name} resulted in status code "${res.status}"`,
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Fetch failed for ${endpoint.name} - ${err.message}`);
+        error = err.message;
+      });
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: status ?? 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  private async pingExternalDependency(
+    endpoint: Endpoint,
+  ): Promise<ExternalDependency> {
+    const pingResult = await ping.promise.probe(endpoint.target);
+
+    let error;
+    if (
+      pingResult.packetLoss === '100.000' ||
+      pingResult.packetLoss === 'unknown'
+    ) {
+      this.logger.error(
+        `Ping failed for ${endpoint.name} - ${pingResult.output}`,
+      );
+      error = pingResult.output === '' ? `${endpoint.target} - Unknown` : pingResult.output;
+    }
+
+    this.logger.info(`Ping results for ${endpoint.name}: ${pingResult.output}`);
+
+    const result: ExternalDependency = {
+      name: endpoint.name,
+      type: endpoint.type,
+      target: endpoint.target,
+      status: pingResult.alive ? 'Healthy' : 'Unhealthy',
+      error: error ?? undefined,
+    };
+
+    return result;
+  }
+
+  public async listConfig(): Promise<JsonValue> {
+    const args = parseArgs(process.argv);
+
+    const configTargets: ConfigTarget[] = [args.config ?? []]
+      .flat()
+      .map(arg =>
+        isValidUrl(arg) ? { url: arg } : { path: resolvePath(arg) },
+      );
+
+    /* eslint-disable-next-line no-restricted-syntax */
+    const paths = findPaths(__dirname);
+
+    const { packages } = await getPackages(paths.targetDir);
+    const schema = await loadConfigSchema({
+      dependencies: packages.map(p => p.packageJson.name),
+    });
```

Comment:
This is actually fairly expensive and doesn't change during runtime. It'd say it's best to memoize the schema.

Reply:
I've known about memoization but never had to do it before - New Skill Unlocked!  🚀